src/game.js

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import {makeShuffledDeck} from "./utils"
+
+class Game extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cardDeck: makeShuffledDeck(),
+      currentCards: [],
+      numOfGames: this.props.games,
+      gameStarted: false,
+      currentRound: 0,
+      currentGame: 1,
+      roundWinner: null,
+      player1CurrScore: 0,
+      player2CurrScore: 0,
+      player1TotalScore: 0,
+      player2TotalScore: 0,
+      
+    };
+  }
+
+  dealCards = () => {
+    const currCards = this.state.cardDeck.slice(-2);
+    if(currCards[0].rank > currCards[1].rank){
+      this.setState({
+        roundWinner: 1,
+        player1CurrScore: this.state.player1CurrScore + 1,
+        player1TotalScore: this.state.player1TotalScore + 1,
+      });
+    }
+    else if (currCards[0].rank < currCards[1].rank){
+      this.setState({
+        roundWinner: 2,
+        player2CurrScore: this.state.player2CurrScore + 1,
+        player2TotalScore: this.state.player2TotalScore + 1,
+      });
+    }
+    else{
+      this.setState({ roundWinner: 0 });
+    }
+      this.setState(
+        (prevState) => ({
+          cardDeck: prevState.cardDeck.slice(0, -2),
+          currentCards: currCards,
+          currentRound: prevState.currentRound + 1,
+          gameStarted: true,
+        }),
+      );
+  };
+
+  nextGame = () => {
+    this.setState((prevState) => ({
+      cardDeck: makeShuffledDeck(),
+      currentCards: [],
+      hasGameStarted: false,
+      currRoundWinner: 0,
+      currentRound: 0,
+      currentGame: prevState.currentGame + 1,
+      player1CurrScore: 0,
+      player2CurrScore: 0,
+    }));
+  };
+
+  resetGame = () => {
+    this.props.getGameMode("start")
```

Comment:
I think we could make this a function that restart the game, without needing any specific input!

Reply:
I did this is to do a full reset, allowing the user to choose the num of games again